compiler/src/dotty/tools/dotc/typer/TypeAssigner.scala

```diff
@@ -315,10 +315,28 @@ trait TypeAssigner {
     }
   }
 
+  /** Substitute argument type `argType` for parameter `pref` in type `tp`,
+   *  skolemizing the argument type if it is not stable and `pref` occurs in `tp`.
+   */
+  def substArgForParam(tp: Type, argType: Type, pref: ParamRef)(implicit ctx: Context) = {
```

Comment:
Minor suggestion: all the `subst*` methods have the `from` argument before the `to` argument, here it's the reverse order which can be a bit confusing. If this is changed it probably makes sense to rename the method to `substParamWithArg` too.